lib/public_suffix.ex

```diff
@@ -59,13 +59,63 @@ defmodule PublicSuffix do
   @spec registrable_domain(String.t, options) :: nil | String.t
   def registrable_domain(domain, options \\ []) when is_binary(domain) do
     # "The registered or registrable domain is the public suffix plus one additional label."
-    parse_domain(domain, options, 1)
+    parse_domain(domain, options, 1, :registrable_or_suffix)
+  end
+
+  @doc """
+  Parses the provided domain and returns the prevailing rule based on the
+  publicsuffix.org rules. If no rules match, the prevailing rule is "*".
+
+  ## Examples
+
+      iex> prevailing_rule("foo.bar.com")
+      "com"
+      iex> prevailing_rule("co.uk")
+      "co.uk"
+      iex> prevailing_rule("foobar.example")
+      "*"
+
+  You can use the `ignore_private` keyword to exclude private (non-ICANN) domains.
+
+      iex> prevailing_rule("foo.github.io", ignore_private: false)
+      "github.io"
+      iex> prevailing_rule("foo.github.io", ignore_private: true)
+      "io"
+      iex> prevailing_rule("foo.github.io")
+      "github.io"
+  """
+  @spec prevailing_rule(String.t) :: nil | String.t
+  @spec prevailing_rule(String.t, options) :: nil | String.t
```

Comment:
Nevermind, I see that `nil` is returned if the input itself is in the invalid form containing a leading dot.  That case is weird and we only support it because the spec mentions it :(.

Do you mind updating the doc to mention when `nil` can be returned since the description (`If no rules match, the prevailing rule is "*"`) makes it sound like `nil` will never be returned?